README.md

```diff
@@ -1,2 +1,307 @@
 # ovn-k-secondary-net-dhcp
-Information on how to deploy a secondary network overlay in Kubernetes for VMs with a DHCP server workload
+Information on how to deploy a secondary network overlay in Kubernetes for VMs
+with a DHCP server workload.
+
+## Introduction
+OVN (Open Virtual Network) is a series of daemons for the Open vSwitch that
+translate virtual network configurations into OpenFlow. It provides virtual
+networking capabilities for any type of workload on a virtualized platform
+(virtual machines and containers) using the same API.
+
+OVN provides a higher-layer of abstraction than Open vSwitch, working with
+logical routers and logical switches, rather than flows. More details can be
+found in the OVN architecture man page.
+
+In this post we will create a secondary network to provide east/west
+connectivity between Openshift Virtualization Virtual Machines. For that, we
```

Comment:
A drawing of the scenario would be great!

Reply:
Let me see how can I provide a neat looking diagram ... 

(I'm out of my league there :stuck_out_tongue_winking_eye: )

Reply:
![image](https://github.com/maiqueb/ovn-k-secondary-net-dhcp/assets/39051226/e3655e3e-9c55-43d6-80ec-e447c2d5ddcd)

Would something this basic be enough ?

Reply:
Hm, do you prefer something like:
![image](https://github.com/maiqueb/ovn-k-secondary-net-dhcp/assets/39051226/1eb1875a-9b73-4620-9b3a-e8155de7a466)

@dominikholler 

I really don't know what to think honestly.

Reply:
Why is the switch below looking different from the one above?
Aren't the VMs also connected to the pod network?

Reply:
I guess that raises questions. The one above is openvswitch based, the one in the bottom a regular switch. 

I would again rather remove all the pod network things. 

I'm willing to remove those from the VM specs if it helps my case :)

Reply:
> It's off scope of this document.

According to my understanding, this blog post is about an alternative to the pod network, so the reader has to understand what the difference is, because this is why he is reading this text.

Reply:
> I guess that raises questions. The one above is openvswitch based, the one in the bottom a regular switch.

Isn't the pod network based on the same openvswitch?

> I would again rather remove all the pod network things.
> I'm willing to remove those from the VM specs if it helps my case :)

No, this is great to highlight the difference. E.g. the pod network could be used to provide the service to the outside, while the internal network might be the connection from the app to the db server.

Reply:
I see it differently. The users wants to have a secondary network for whatever reasons their use cases need, and I present an alternative that doesn't require nmstate / provisioning VLANs which is easier to configure. That setup is only valid if the use cases support east / west traffic exclusively.

Reply:
>  That setup is only valid if the use cases support east / west traffic exclusively.

exectly, and this is why it would be great if the diagram would reflect this.

> I present an alternative that doesn't require nmstate / provisioning VLANs which is easier to configure.

Is there already a blog post or another likable text, picture, video or slides which explains how to do this, so that the reader without knowledge get the hint to dive in?

Reply:
this one: https://kubevirt.io/2020/Multiple-Network-Attachments-with-bridge-CNI.html

... which is not a redhat cloud blog. 

I would like to make it clear I see value in what you propose - adding a "why" section indicating the differences between the pod network and secondary networks, what comes for free on the pod network, the cost of setting up secondaries and so forth, but I feel that's against the cloud blog guidelines. 

IMO we should tackle a single thing here, and keep the content small, concise, and focused on the task at hand - which is to configure secondary network overlays using this technology. 

I do not mind collaborating on getting a "generic" network guidelines for openshift virtualization out there, but that would serve a different purpose - one of which would be to educate the customer on what choices are out there, and what network fabrics are better suited to the task.

Reply:
> 
> this one: https://kubevirt.io/2020/Multiple-Network-Attachments-with-bridge-CNI.html
> 
> ... which is not a redhat cloud blog. 
> 

never mind, it would be a nice to have to add it as a link to the "does
not require provisioning VLANs or any other physical network resources"

> I would like to make it clear I see value in what you propose - adding a "why" section indicating the differences between the pod network and secondary networks, what comes for free on the pod network, the cost of setting up secondaries and so forth, but I feel that's against the cloud blog guidelines. 
> 
> IMO we should tackle a single thing here, and keep the content small, concise, and focused on the task at hand - which is to configure secondary network overlays using this technology. 
> 
> I do not mind collaborating on getting a "generic" network guidelines for openshift virtualization out there, but that would serve a different purpose - one of which would be to educate the customer on what choices are out there, and what network fabrics are better suited to the task. 
> 
> 

No round up on all networking options available in this blog post. I just think it might be helpful to let the reader know, which problem the solution solves.

Reply:
@maiqueb Thanks, the new diagram looks great, from my point of view!